### DISCUSSION TITLE
How to manange deployments with different FpConfig.h or settings.ini needs

### DISCUSSION THREAD

**Discussion by SterlingPeet** [nasa/fprime#2846] [Q&A]
I am working on an example deployment repository that ultimately demonstrates a few different configurations on the same target hardware for reference purposes.  Primarily I need to be able to provide references where the opcodes for comands/telemetry/events are different word widths.  I know this has a GDS implication, which will be described in the README.

My question is this: How can I reasonably pack these into the same github repository?

If there is a clean and supported way to adjust the settings.ini or FpConfig.h files from the deployment ``CMakeLists.txt`` folder that would be ideal.  Otherwise I can write logic in cmake that overwrites this file for the current deployment build.  I am not fond of that solution, but it should work for demonstration purposes.  The swapping behavior would be documented in the README for people looking to replicate the build properties in their own project.

**Comment by tbkevin** [comment]
Hi, I'm facing a related issue regarding this matter. I need to upload my project on a github repository, but I've added some modifications on the " FpConfig.h" file, "TlmChanImplCfg.hpp" and Linux Timer Component.
Did you manage to figure out a proper way? Or you just used Cmake? And if so, could you give some hints on the cmakes solution, because I'm newbie to Cmake too.

Thanks in advanced.

**Reply by thomas-bc** [reply]
@tbkevin you need to copy the `config/` directory to the root of your project:

````
cp -r MyProject/fprime/config MyProject/
````

Then change the settings.ini file to point to this location

We have a reference repo that follows this pattern here (notice the value for ./config in the settings.ini file): 
https://github.com/fprime-community/fprime-baremetal-reference/blob/main/settings.ini
